packages/ui/src/machines/authenticator/signUp.ts

```diff
@@ -242,7 +242,15 @@ export function createSignUpMachine({ services }: SignUpMachineOptions) {
         setCodeDeliveryDetails,
         setUser,
         sendUpdate: sendUpdate(), // sendUpdate is a HOC
-        setAutoSignInIntent: assign({ intent: (_) => 'autoSignIn' }),
+        setAutoSignInIntent: assign({
+          intent: (context) => {
+            if (context?.intent === 'confirmSignUp') {
+              return 'autoSignInSubmit';
+            } else {
+              return 'autoSignIn';
+            }
+          },
+        }),
```

Comment:
nit:

```suggestion
        setAutoSignInIntent: assign({
          intent: (context) => context?.intent === 'confirmSignUp' ? 'autoSignInSubmit': 'autoSignIn',
        }),
```

Reply:
I prefer to not set this as ternary for now.

Reply:
I'm going for more readability. The ternary is a bit more concise though.